rules/aws_resource_missing_tags.go

```diff
@@ -77,29 +150,67 @@ func (r *AwsResourceMissingTagsRule) Check(runner tflint.Runner) error {
 		}
 
 		resources, err := runner.GetResourceContent(resourceType, &hclext.BodySchema{
-			Attributes: []hclext.AttributeSchema{{Name: tagsAttributeName}},
+			Attributes: []hclext.AttributeSchema{
+				{Name: tagsAttributeName},
+				{Name: providerAttributeName},
+			},
 		}, nil)
 		if err != nil {
 			return err
 		}
 
+		if resources.IsEmpty() {
+			continue
+		}
+
 		for _, resource := range resources.Blocks {
-			if attribute, ok := resource.Body.Attributes[tagsAttributeName]; ok {
-				logger.Debug("Walk `%s` attribute", resource.Labels[0]+"."+resource.Labels[1]+"."+tagsAttributeName)
-				wantType := cty.Map(cty.String)
-				err := runner.EvaluateExpr(attribute.Expr, func(resourceTags map[string]string) error {
+			providerName := "default"
+			if val, ok := resource.Body.Attributes[providerAttributeName]; ok {
+        if diagnostics := gohcl.DecodeExpression(val.Expr, nil, &providerName); diagnostics.HasErrors() {
```

Comment:
This fix is not enough. Presumably, tests for attributes like `provider = aws.west` should fail.
Probably the easiest fix is to expose the `decodeProviderConfigRef` and make use of it.
https://github.com/terraform-linters/tflint-ruleset-aws/blob/409b25015a71c10a83c4bcd449f0524abff79a91/aws/decode.go#L28

Reply:
Made DecodeProviderConfig and fixed the references, ty!